service/verify.go

```diff
@@ -0,0 +1,170 @@
+package main
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/slsa-framework/slsa-verifier/cmd"
+	"github.com/slsa-framework/slsa-verifier/verification"
+)
+
+var errInvalid = errors.New("invalid")
+
+type v1Query struct {
+	// Compulsory fields.
+	Source       string `json:"source"`
+	ArtifactHash string `json:"artifactHash"`
```

Comment:
for containers, we may want to support this as a oneof artifacthash/dsse envelope or container image reference.

I would rather see grpc service like definitions, but save for a later review. just commenting on the idea for now.

Reply:
SG.